src/gui/addnewtorrentdialog.cpp

```diff
@@ -241,7 +238,7 @@ void AddNewTorrentDialog::show(const QString &source, const BitTorrent::AddTorre
         : dlg->loadTorrentFile(source);
 
     if (isLoaded)
-        dlg->open();
+        dlg->QDialog::show();
```

Comment:
Is fully qualified name really required here?

Reply:
yes, compiler complains about it if not.

Reply:
This is weird... Can you give me an example of a compiler message?

Reply:
From gcc 8.3.0
```
../../src/gui/addnewtorrentdialog.cpp: In static member function ‘static void AddNewTorrentDialog::show(const QString&, const BitTorrent::AddTorrentParams&, QWidget*)’:
../../src/gui/addnewtorrentdialog.cpp:241:19: error: no matching function for call to ‘AddNewTorrentDialog::show()’
         dlg->show();
                   ^
../../src/gui/addnewtorrentdialog.cpp:223:6: note: candidate: ‘static void AddNewTorrentDialog::show(const QString&, const BitTorrent::AddTorrentParams&, QWidget*)’
 void AddNewTorrentDialog::show(const QString &source, const BitTorrent::AddTorrentParams &inParams, QWidget *parent)
      ^~~~~~~~~~~~~~~~~~~
../../src/gui/addnewtorrentdialog.cpp:223:6: note:   candidate expects 3 arguments, 0 provided
In file included from ../../src/gui/addnewtorrentdialog.cpp:29:
../../src/gui/addnewtorrentdialog.h:79:17: note: candidate: ‘static void AddNewTorrentDialog::show(const QString&, QWidget*)’
     static void show(const QString &source, QWidget *parent);
                 ^~~~
../../src/gui/addnewtorrentdialog.h:79:17: note:   candidate expects 2 arguments, 0 provided
```

Reply:
Oh, right, the base class methods are not considered when resolving names. So you either need to use fully qualified name here or add "using QDialog::show;" in class declaration.

Reply:
Unless you prefer the other way, I would like to settle on using fully qualified name.
I think `using QDialog::show;` & `dlg->show();` in a static `show()` function is not easy to see that we are calling the base `show()`.

Reply:
Ok, let it be as is.